libgnucash/core-utils/gnc-locale-utils.c

```diff
@@ -164,10 +164,8 @@ gchar *gnc_locale_name (void)
 int
 gnc_collation_compare (const char* a, const char* b)
 {
-    char* akey = a ? g_utf8_collate_key (a, -1) : NULL;
-    char* bkey = b ? g_utf8_collate_key (b, -1) : NULL;
-    int result = g_strcmp0 (akey, bkey);
-    if (akey) g_free (akey);
-    if (bkey) g_free (bkey);
-    return result;
+    if (!a && !b) return 0;
+    if (!a) return -1;
+    if (!b) return 1;
```

Comment:
This is a good place for the ternary operator:
```C
if (!(a && b))
   return a ? 1 : b ? -1 : 0;
```
or even
```C
return !a ? (!b ? 0 : -1) : !b ? 1 : g_utf8_collate (a, b);
```

Reply:
or, avoid negation:
```c
    return a ? (b ? g_utf8_collate (a, b) : 1) : b ? -1 : 0;
```